source/Engine.cpp

```diff
@@ -2327,8 +2326,13 @@ void Engine::DoWeather(Weather &weather)
 		// Get all ship bodies that are touching a ring defined by the hazard's min
 		// and max ranges at the hazard's origin. Any ship touching this ring takes
 		// hazard damage.
-		for(Body *body : (hazard->SystemWide() ? shipCollisions.All()
-			: shipCollisions.Ring(weather.Origin(), hazard->MinRange(), hazard->MaxRange())))
+		vector<Body *> affectedShips;
+		affectedShips.reserve(ships.size());
```

Comment:
ships.size() seems fine for hazards, but line 2332 overrides the vector, so I would expect the reserve to only happen in the else part.